front/lib/api/assistant/generation.ts

```diff
@@ -70,46 +70,73 @@ export async function renderConversationForModelMultiActions({
       const actions = removeNulls(m.actions);
 
       // This array is 2D, because we can have multiple calls per agent message (parallel calls).
-      const steps = [] as Array<
-        Array<{ call: FunctionCallType; result: FunctionMessageTypeModel }>
-      >;
+
+      const steps = [] as Array<{
+        content: string | null;
```

Comment:
Shouldn't this be an array to render an array of text content?

Reply:
That's a fair point

Reply:
Actually, no it's not possible, because we emit the actions at the very last step of the step. We have to, because we only get the function calls after we are done running the dust app. So the content will always be "before" the actions